primer/src/Primer/Eval/NormalOrder.hs

```diff
@@ -129,12 +131,17 @@ data RedexWithContext
   | RType TypeZ RedexType
 
 viewLet :: (Dir, ExprZ) -> Maybe (LetBinding, Accum Cxt (Dir, ExprZ))
-viewLet dez@(_, ez) = case (target ez, exprChildren dez) of
+viewLet dez@(_, ez) = case (target ez, exprChildren' dez) of
   (Let _ x e _b, [_, bz]) -> Just (LetBind x e, bz)
   (Letrec _ x e ty _b, [_, bz]) -> Just (LetrecBind x e ty, bz)
   (LetType _ a ty _b, [bz]) -> bz `seq` Just (LetTyBind $ LetTypeBind a ty, bz)
   _ -> Nothing
 
+data NormalOrderOptions -- TODO/REVIEW: this is a bit of a misnomer, as it affects `Eval.redexes`, which is only used for single-step. NB: even if we set StopAtBinders, we can still call `Eval.step` with the id of a redex under a binder and it will reduce, even though `redexes` will not report that id
+  = UnderBinders
+  deriving stock (Eq, Show, Read, Generic)
+  deriving (FromJSON, ToJSON) via PrimerJSON NormalOrderOptions
```

Comment:
Thoughts? I would like either a better name, or to decide that `redexes` will always allow eval under binders. Probably worth thinking about what API consumers would prefer here.

Reply:
Unless I misunderstand this option, I don't think we'll want to prevent eval under binders in step-eval.

Reply:
The option can be set per API call. So the question is "should step-eval be hardcoded to one choice (either allow eval under binders or not), or should the consumer decide?". (Actually, the way it is implemented is `redexes` has the option, and simply reports which nodes are redexes; a frontend is expected to ask for one of these to be reduced in a separate API call, and this call is ignorant of the option -- technically the frontend can ask for redexes-disallowing-eval-under-binders, and then request to reduce some other node which happens to be under a binder and it will just work. However, this is a somewhat pointless technical aside!)

Reply:
I can't come up with any reasons why we'd want to disable eval under binders in step eval. Am I missing anything?

Reply:
This option is only currently intended to prevent runaway eval of top-level functions in the eval window. I feel like we’re talking past each other because I don’t think understand why you think it would have any value in step mode. Let’s discuss on our next face-to-face meaning so we don’t go in circles here.

Reply:
We decided that step eval should always be able to go under binders. We feel that it is less confusing to allow students to evaluate every redex at the expense of full-eval consistency. (Recall that the impetus for this PR is to avoid run-away inlining of recursive top-level-definitions, rather than a philosophical stance on what a "normal form" should be)

TODO
- [ ] Implement this change